swagger/formats.go

```diff
@@ -0,0 +1,35 @@
+package swagger
+
+import (
+	"regexp"
+
+	"github.com/go-openapi/strfmt"
+)
+
+func InitCustomFormats() {
+	m := mongoID("")
+	strfmt.Default.Add("mongo-id", &m, isMongoID)
+}
+
+var mongoRegExp = regexp.MustCompile("^[0-9a-f]{24}$")
+
+func isMongoID(str string) bool {
+	return mongoRegExp.MatchString(str)
```

Comment:
after reading the code in bson.go and `hex.Decode`, I now think the regexp check is fine